### PR TITLE
Support Slashes in Feature Names in UI/API

### LINKED ISSUE #357: Flipper UI crashes if feature key contains a '/'
How to reproduce:

Step 1 : create a key in the console

```
  bin/rails c
  > Flipper.enabled? 'a/b'
    => false
  > Flipper.enable 'a/b'
    => true
  > Flipper.enabled? 'a/b'
    => true
```

Step 2 : try and edit the feature in flipper ui
```
 open https://mydomain.com/features/features
 click on the 'a/b' key
 -> https://mydomain.com/features/features/a/b
                                           ^^^
```
result (in the browser)
```
  Flipper::UI::Actions::Gate does not support request method "get"
```

### PR DESCRIPTION
**Pull request by jnunemaker**
As brought up in https://github.com/jnunemaker/flipper/issues/357, flipper UI (and API) do not handle slashes in feature names. This fixes that. I had to redo routing stuff, but it should all work the same.

I considered adding a note that slashes aren't supported, but it wasn't too bad to add support so why not. There is some duplication in finding the feature name, but I didn't feel like there was enough to warrant any abstracting just yet. We'll see if this churns at all and would benefit from that down the road.

fixes #357

### CODE REVIEW

File: spec/flipper/ui/actions/actors_gate_spec.rb [hunk 1]
```diff
@@ -27,6 +27,21 @@
...
+    end
+
+    it 'responds with success' do
+      expect(last_response.status).to be(200)
+    end
+
+    it 'renders add new actor form' do
+      form = '<form action="/features/a/b/actors" method="post" class="form-inline">'
+      expect(last_response.body).to include(form)
+    end
```
Comment (AlexWheeler) [inline spec/flipper/ui/actions/actors_gate_spec.rb]: 👍

File: lib/flipper/api/v1/actions/actors_gate.rb [hunk 2]
```diff
@@ -33,7 +34,10 @@ def ensure_valid_params
           end
 
           def feature_name
-            @feature_name ||= Rack::Utils.unescape(path_parts[-2])
+            @feature_name ||= begin
+              match = request.path_info.match(REGEX)
+              match ? match[1] : nil
```
[the remark this answers is not in the document]
Comment (jnunemaker) [reply lib/flipper/api/v1/actions/actors_gate.rb]: I almost went for a capture group, but this took so much longer than I was planning I lost steam. I think it is worth updating to use them so I'll spin back around on it soon.

[the remark this answers is not in the document]
Comment (jnunemaker) [reply lib/flipper/api/v1/actions/actors_gate.rb]: Updated in f61bbe9

**Comment by jnunemaker** [comment]
Moved back to single regex in 8fa06a54d0438794124845f35e89e54f8ac534a1. Going to get rid of the constants too.

**Comment by jnunemaker** [comment]
Ok, waiting for CI but I think this is ready and I'm going to merge soon.

**Pull request merged**